cmds/decpu/decpu.go

```diff
@@ -207,13 +209,23 @@ func main() {
 	cpus := make(chan *cpu, 1)
 
 	if err == nil {
-		sdHost, sdPort, err := ds.Lookup(dq)
-		if err == nil {
-			cpus <- &cpu{host: sdHost, port: sdPort}
-			close(cpus)
-		} else {
-			verbose("ds.Lookup returned %w", err)
-		}
+		go func() {
+			defer close(cpus)
+			c, cancel := ds.LookupChan(dq)
+			defer cancel()
+			var amt int
+			for e := range c {
+				amt++
+				if amt > *numCPUs {
+					return
```

Comment:
okay, so if we were to move sorting to the client, instead of instantiating the cpus on every loop, we wait till we hit n, sort the list and then instantiate.  I suppose the downside is with the code the way it is we dispatch as soon as the remote tells us and otherwise we are waiting for our timeout before dispatching.  We could conditionalize holding off on dispatching ONLY if we detect sort -- but we'd need parse to return a bool telling us whether or not there is a sort involved (by default there is though).